### ISSUE TITLE
extention ram explotion

### ISSUE THREAD

**Issue by alidan** [ccd0/4chan-x#2618]
![taskmgr_2020-05-31_05-47-28](https://user-images.githubusercontent.com/2681758/83350531-533d2680-a302-11ea-8643-3436fa33bb17.png)

I think this is because of the extension, larger threads and clicking replies will sometimes just cause the tab to freeze up, I left this one to see if it would recover but no, chrome killed it after it because I estimate 12-20gb of ram large

not 100% sure if its 4chanx, but as it only happens with 4chan, its where Im starting to look.

**Comment by yejip79190** [comment]
I have the same issue. My CPU and memory usage goes up to like 50% and when I look at the Chrome task manager it's always showing the 4chan tabs as using the biggest amount of memory.

**Comment by alidan** [comment]
for me its individual tabs, they are more or less the same till ram bloats to hell and chrome kills the tab, though as you can see, I only have around 10-14gb of programs in the background working, the rest is chrome

**Comment by ccd0** [comment]
Would you mind posting a copy of your exported settings? (Edit out any sensitive info such as tripcode passwords.)

**Comment by alidan** [comment]
https://www.mediafire.com/file/igmhexx5kbd3dsu/4chan_X_v1.14.19.3-1590951964051.json/file

Now that you mention
_(Edit out any sensitive info such as tripcode passwords.)_
would it be possible to have a 'sanitized' export that would scrub the export of any info that could be sensitive or potentially identifiable?

**Comment by ccd0** [comment]
It's a good idea, but what's considered sensitive might vary from person to person.

**Comment by ccd0** [comment]
Imported your settings and still having trouble reproducing. Are you using the extension or the userscript? Are you using the same version of Chrome as https://github.com/ccd0/4chan-x/issues/2613#issuecomment-636387218 ? Do you have any other add-ons installed? Are there particular threads you notice this in?

**Comment by alidan** [comment]
I have 4chan x as an extension, alongside an exlinks script
https://boards.4channel.org/v/thread/510401390
that thread and just open every reply
![chrome_2020-05-31_17-00-43](https://user-images.githubusercontent.com/2681758/83363611-56ff9600-a360-11ea-9bce-1d6b375b3e9f.png)
for me it locked up and bloated shortly after I got passed the op reply chains

as for user identifiable, yea, some people may not care at all, but I opened mine in notepad++ and I have absolutely no idea where I would even start to sanitize it. 
possibly with a sanitized export, a popup could ask you what you want to leave out, I can really only think of threads you posted in, tripcodes, filters, and watchers as things some people may want to not share.

**Comment by ccd0** [comment]
![2020-05-31-160550_1919x980_scrot](https://user-images.githubusercontent.com/5894749/83364799-a3df6e80-a358-11ea-965e-083e696d47a2.png)
I inlined a large number of backlinks and I'm still not seeing this. Does it happen in a fresh profile?
https://developer.chrome.com/devtools/docs/clean-testing-environment

**Comment by roboxa** [comment]
Made an account just to say I have these same problems no matter what kind of instance of chrome I try. Blank profile, incognito mode, existing profile, etc they all have the same results. 4chan has big performance issues and eats a ton of CPU usage when doing things like posting images.

I tried to clear my thread watcher to see if that would fix it and it completely froze the page and chrome automatically closed it. So the only thing I could do was a reset of the settings. 

Is it possible to download an old version of 4chan X for testing? This wasn't happening too long ago and I know its gotten updated recently.

**Comment by ccd0** [comment]
Old versions are in the changelog:
https://github.com/ccd0/4chan-x/blob/master/CHANGELOG.md

**Comment by alidan** [comment]
i'm going to opt out of doing the blank profile test till I see this happen more often, The problem is I have had it happen I think 4 times in 8 days, and each time it happened it was relatively random so I can't really force it, each time it happened it was when I was going down the threads, its possible that it was something to do with going down the threads and highlighting parts of  the posts, i'm not sure, next time it happens ill take a screenshot of exactly what the window looks like when it happened. 

more or less, if I can't force the problem with my current profile, then it would be completely random chance for it to happen on a sanitized one and I wouldnt know if there was something underlining the issue. 

so yea, if it happens, ill post a screenshot of what I did, if it doesn't, then it was a bug in chrome or extention that worked itself out.

**Comment by ccd0** [comment]
Fair enough. Make sure your ad blocker is turned on for 4chan too, because otherwise you could be getting ads or ad-related scripts that cause this sort of thing.

**Comment by alidan** [comment]
lol, 4chan is the sole reason I got and deal with umatrix, got to be able to block all the bs, not just what the adblockers know about.
